---
title: "Effects of N Addition on AMF Diversity in WI Switchgrass Cropping Systems"
subtitle: "All Sites"
author: "Alden Dirks"
date: "7/26/2019"
output: html_document
---

```{r setup, include=FALSE}
rm(list = ls()) # clean up environment for new session
knitr::opts_knit$set(root.dir = '..' ) # set markdown root directory to project root
knitr::opts_chunk$set(echo = TRUE, fig.path = '../figures/', message = FALSE, warning = FALSE)
```

```{r libraries, results = 'hide'}
library("Biostrings"); packageVersion("Biostrings") # version 2.52.0
library("DESeq2"); packageVersion("DESeq2") # version 1.24.0
library("gridExtra"); packageVersion("gridExtra") # version 2.3
library("lme4"); packageVersion("lme4") # version 1.1.21
library("lmerTest"); packageVersion("lmerTest") # version 3.1.0
library("party"); packageVersion("party") # version 1.3.3
library("phyloseq"); packageVersion("phyloseq") # version 1.28.0
library("randomForest"); packageVersion("randomForest") # version 4.6.14
library("rpart"); packageVersion("rpart") # version 4.1.15
library("tidyverse"); packageVersion("tidyverse") # version 1.2.1
```

```{r paths}
path_data = file.path("data", "seqs", "cleaned")
path_master = file.path("data")
```

```{r load_data}
asv <- read.table(file.path(path_data, "ASVs_counts.csv"), header = TRUE, check.names = FALSE, sep = ",")
fas <- readDNAStringSet(file.path(path_data, "ASVs.fasta"))
info <- read.table(file.path(path_data, "sample_info.csv"), header = TRUE, check.names = FALSE, sep = ",")
info$block <- info$site:info$block
master <- read.table(file.path(path_master, "master.csv"), header = TRUE, check.names = FALSE, sep = ",")
tax <- as.matrix(read.table(file.path(path_data, "ASVs_taxonomy.csv"), header = TRUE, row.names = 1, check.names = FALSE, sep = ","))
```

## ASV Correlation Matrix

Which ASVs are present in the exact same samples? 

Noninformative groupings:

* 39 ASVs did not occur in the same pattern of samples as any other ASV
* 341 ASVs only occurred in one sample, with other ASVs also only occurring in that sample

Informative groupings:

* 34 ASVs occurred in more than one sample in the same pattern as one or more other ASVs

Groups:

* ASVs 8, 47 - Paraglomus
  - differ in two subsequent nucleotides (TC vs. AT)
  - BLAST reports 87.50% identity with KF849540
* ASVs 10, 14 - Paraglomus
  - a few SNPs
  - BLAST reports close affinity to isolate ATT260-8 clone pCK084-12 (FR750048)
* ASVs 24, 32 - Paraglomus
* ASVs 45, 59 - Paraglomus
* ASVs 53, 115 - Paraglomus
* ASVs 56, 150 - Paraglomus (1) and Scutellospora (1)
* ASVs 94, 123 - Dominikia
* ASVs 13, 29, and 66 - Paraglomus
* ASVs 37, 57, 176, 246 - Dominikia (3) and Septoglomus (1)
* ASVs 28, 42, 80, 91, 143 - Paraglomus (2), Dominikia (1), Paraglomus (2)
* ASVs 25, 27, 41, 44, 81, 84, 104, 231 - Dominikia (6), Paraglomus (1), Dominikia (1)
  - ASVs 104 and 231 do not seem to be complete sequences 

```{r}
# for each column, mark as TRUE if present or FALSE if not
asv_bool <- apply(master[,28:439], 2, as.logical)

# create empty list
asv_list <- list()

# for each column, go through the asv_bool dataframe and determine which columns are equal
#   (i.e., which ASVs occur in the same samples only)
for (i in seq(1:dim(asv_bool)[2])) {
  asv_list[i] <- list(apply(asv_bool, 2, function(x) all(asv_bool[,i] == x)))
}
#asv_list[[1]]

# rename list
names(asv_list) <- colnames(asv_bool)

# sum each vector to determine how many ASVs occur in the same samples 
asv_list_sum <- lapply(asv_list, function(x) sum(unname(x)))
asv_vector <- unlist(asv_list_sum)
table(asv_vector) 
sum(table(asv_vector))
# if 1, then only matched itself in samples that it occurred in 
# if 34, then 34 different ASVs ocurred in the same samples 

# make a vector of the different groupings of numbers of ASVs that occur in the same samples
asv_numbers <- as.numeric(names(table(asv_vector)))
asv_numbers

# 1
tax[rownames(tax) %in% names(which(asv_list_sum == 1)),]
master[,colnames(master) %in% names(which(asv_list_sum == 1))]

# 2 
# 8 sets of pairs of ASVs that occur in the same samples
# 1 pair only occur in one sample together (Rhinelander R2 amended)
tax[rownames(tax) %in% names(which(asv_list_sum == 2)),]
master[,colnames(master) %in% names(which(asv_list_sum == 2))]

# 3 
# 2 sets of three ASVs that occur in the same samples
# 1 set is a group of ASVs taht only occur in one sample
tax[rownames(tax) %in% names(which(asv_list_sum == 3)),]
master[,colnames(master) %in% names(which(asv_list_sum == 3))]

# 4 
# 2 sets of four ASVs that occur in the same samples
# 1 set is a group of ASVs that only occur in one sample
tax[rownames(tax) %in% names(which(asv_list_sum == 4)),]
master[,colnames(master) %in% names(which(asv_list_sum == 4))]

# 5
# 3 sets
# 2 of the three sets are groups of ASVs that only occur in one sample together
tax[rownames(tax) %in% names(which(asv_list_sum == 5)),]
master[,colnames(master) %in% names(which(asv_list_sum == 5))]

# 8 - rows 15 AND 16
# 7 of the 8 are Dominikia indica, one is Paraglomus brasilianum
tax[rownames(tax) %in% names(which(asv_list_sum == 8)),]
master[,colnames(master) %in% names(which(asv_list_sum == 8))]

## Singletons (sets of ASVs that only occured in the same one sample)

# 6
tax[rownames(tax) %in% names(which(asv_list_sum == 6)),]
master[,colnames(master) %in% names(which(asv_list_sum == 6))]

# 7
tax[rownames(tax) %in% names(which(asv_list_sum == 7)),]
master[,colnames(master) %in% names(which(asv_list_sum == 7))]

# 10 - row 3
tax[rownames(tax) %in% names(which(asv_list_sum == 10)),]
master[,colnames(master) %in% names(which(asv_list_sum == 10))]

# 13 - rows 1, 16, or 8
tax[rownames(tax) %in% names(which(asv_list_sum == 13)),]
master[,colnames(master) %in% names(which(asv_list_sum == 13))]

# 16 - all the ASVs that only occurred in Oregon R4 control
tax[rownames(tax) %in% names(which(asv_list_sum == 16)),]
master[,colnames(master) %in% names(which(asv_list_sum == 16))]

# 19 (two sets) - all the ASVs that only occurred in Oregon R2 amended or Oregon R2 control
tax[rownames(tax) %in% names(which(asv_list_sum == 19)),]
master[,colnames(master) %in% names(which(asv_list_sum == 19))]

# 21 - all the ASVs that only occurred in Oregon R3 control
tax[rownames(tax) %in% names(which(asv_list_sum == 21)),]
master[,colnames(master) %in% names(which(asv_list_sum == 21))]

# 22 (two sets) - all the ASVs that only occurred in Hancock R4 amended or Hancock R3 control
tax[rownames(tax) %in% names(which(asv_list_sum == 22)),]
master[,colnames(master) %in% names(which(asv_list_sum == 22))]

# 24 - all the ASVs that only occurred in Oregon R1 amended
tax[rownames(tax) %in% names(which(asv_list_sum == 24)),]
master[,colnames(master) %in% names(which(asv_list_sum == 24))]

# 26 (two sets) - all the ASVs that only occurred in Oregon R4 amended or Oregon R4 amended
tax[rownames(tax) %in% names(which(asv_list_sum == 26)),]
master[,colnames(master) %in% names(which(asv_list_sum == 26))]
master[,colnames(master) %in% names(which(asv_list_sum == 26))][11,] > 0
master[,colnames(master) %in% names(which(asv_list_sum == 26))][13,] > 0

# 29 - all the ASVs that only occurred in Hancock R4 control
tax[rownames(tax) %in% names(which(asv_list_sum == 29)),]
master[,colnames(master) %in% names(which(asv_list_sum == 29))]

# 34 - all the ASVs that only ocurred in Hancock R2 control
tax[rownames(tax) %in% names(which(asv_list_sum == 34)),]
master[,colnames(master) %in% names(which(asv_list_sum == 34))]
```




Test taxon - check out Rhizophagus intraradices, go to the original genome study and see what the ASVs are, see how they correlate with each other in my study

```{r}
cor_mat <- cor(master[,28:50], method = "pearson", use = "complete.obs")
round(cor_mat, 2)
abs(cor_mat) > .9
master$ASV_2
master$ASV_30
3187/963
```

## Visualization of ASVs

```{r all_sites_pie_charts}
ggplot(as.data.frame(tax), aes(x = factor(1), fill = Genus)) +
  geom_bar(width = 1) +
  xlab("") + ylab("") +
  coord_polar("y")
ggplot(as.data.frame(tax), aes(x = factor(1), fill = Family)) +
  geom_bar(width = 1) +
  xlab("") + ylab("") +
  coord_polar("y")
ggplot(as.data.frame(tax), aes(x = factor(1), fill = Order)) +
  geom_bar(width = 1) +
  xlab("") + ylab("") +
  coord_polar("y")
```

```{r all_sites_box_plots}
ggplot(master, aes(x = site, y = div_rich, fill = nitrogen)) + geom_boxplot() + labs(y = "ASV Richness", x = "Site")
ggplot(master, aes(x = site, y = div_shan, fill = nitrogen)) + geom_boxplot() + labs(y = "ASV Shannon Index", x = "Site")
ggplot(master, aes(x = site, y = div_simp, fill = nitrogen)) + geom_boxplot() + labs(y = "ASV Simpson's Index", x = "Site")
```

## Classification and Regression Tree Analysis

```{r cart}
rt1 <- rpart(div_rich ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master, method = "anova")
printcp(rt1)
plotcp(rt1)
summary(rt1)
rt2 <- rpart(div_shan ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master, method = "anova")
printcp(rt2)
plotcp(rt2)
summary(rt2)
rt3 <- rpart(div_simp ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master, method = "anova")
printcp(rt3)
plotcp(rt3)
summary(rt3)
rt4 <- rpart(yield ~ site + nitrogen + div_rich + div_shan + div_simp + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master, method = "anova")
printcp(rt4)
plotcp(rt4)
summary(rt4)
```

## Conditional Inference Trees

```{r conditional_inference}
ct1 <- ctree(yield ~ site + nitrogen + div_rich + div_shan + div_simp + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master)
plot(ct1)
ct2 <- ctree(div_rich ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master)
plot(ct2)
ct3 <- ctree(div_shan ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master)
plot(ct3)
ct4 <- ctree(div_simp ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master)
plot(ct4)
```

## Random Forest

```{r random_forest}
rf1 <- randomForest(yield ~ site + nitrogen + div_rich + div_shan + div_simp + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master)
print(rf1)
importance(rf1)
rf2 <- randomForest(div_rich ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master)
print(rf2)
importance(rf2)
rf3 <- randomForest(div_shan ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master)
print(rf3)
importance(rf3)
rf4 <- randomForest(div_simp ~ site + nitrogen + ph + p_ppm + est_cec + per_N + per_C + cn_ratio, data = master)
print(rf4)
importance(rf4)
```

## Linear Mixed Effects Regression Analysis

* Biomass differs significantly by site (p = 0.0002464) and by N treatment (p = 0.0209347). There is no interaction (p = 0.1395920). Biomass across the sites is as follows: Oregon > Rhinelander > Hancock, as we would expect given our knowledge of the soil fertility of each site. N amendment increases biomass.
* There is a significant site*nitrogen interaction for yield (p = 0.005733). Nitrogen amendment does not significantly affect yield at Hancock (p = 0.500882, B corrected p = 1.0) or Oregon (p = 0.071604, B corrected p = 0.214812) but it signiicantly increases it at Rhinelander (p = 0.002268, B corrected p = 0.006804).
* Plant height in September did not differ by site or nitrogen treatment and there was no interaction.
* Lodging frequency differed by site (p = 0.001885) with Oregon having a greater frequency of lodging than Rhinelander and Hancock. 

```{r lmer_agr, results = 'hide', eval = FALSE}
lmer_biomass <- lmer(biomass ~ site*nitrogen + (1|block:site), data = master, REML = FALSE); summary(lmer_biomass); anova(lmer_biomass); difflsmeans(lmer_biomass)
lmer_yield <- lmer(yield ~ site*nitrogen + (1|block:site), data = master); anova(lmer_yield); difflsmeans(lmer_yield)
lmer_height <- lmer(height ~ site*nitrogen + (1|block:site), data = master); anova(lmer_height); difflsmeans(lmer_height)
lmer_lodged <- lmer(lodged ~ site*nitrogen + (1|block:site), data = master); anova(lmer_lodged); difflsmeans(lmer_lodged)
```

* pH, percent C, percent N, C:N ratio, and cation exchange capacity differ significantly by site. 
* There was actually a significant site by nitrogen interaction for phosphorus (p = 0.04241). Rhinelander and Oregon did not differ significantly with the addition of N, but Hancock did (p = 0.0214789). However, when corrected for multiple comparisons, this was no longer significant (p = 0.0644367).

```{r lmer_soil, results = 'hide', eval = FALSE}
lmer_ph <- lmer(ph ~ site*nitrogen + (1|block:site), data = master); anova(lmer_ph); difflsmeans(lmer_ph)
lmer_p <- lmer(p_ppm ~ site*nitrogen + (1|block:site), data = master); anova(lmer_p); difflsmeans(lmer_p)
lmer_n <- lmer(per_N ~ site*nitrogen + (1|block:site), data = master); anova(lmer_n); difflsmeans(lmer_n)
lmer_c <- lmer(per_C ~ site*nitrogen + (1|block:site), data = master); anova(lmer_c); difflsmeans(lmer_c)
lmer_cn <- lmer(cn_ratio ~ site*nitrogen + (1|block:site), data = master); anova(lmer_cn); difflsmeans(lmer_cn)
lmer_cec <- lmer(est_cec ~ site*nitrogen + (1|block:site), data = master); anova(lmer_cec); difflsmeans(lmer_cec)
```

* rarefied richness and richness differ by site (p = 0.0107) as follows: Oregon and Hancock > Rhinelander.
* diveristy did not differ by site or N treatment. 

```{r lmer_div, results = 'hide', eval = FALSE}
lmer_rar <- lmer(div_rar ~ site*nitrogen + (1|block:site), data = master); anova(lmer_rar); difflsmeans(lmer_rar)
plot(lmer_rar)
lmer_rich <- lmer(div_rich ~ site*nitrogen + (1|block:site), data = master); anova(lmer_rich); difflsmeans(lmer_rich)
plot(lmer_rich)
lmer_shan <- lmer(div_shan ~ site*nitrogen + (1|block:site), data = master); anova(lmer_shan); difflsmeans(lmer_shan)
plot(lmer_shan)
lmer_simp <- lmer(div_simp ~ site*nitrogen + (1|block:site), data = master); anova(lmer_simp); difflsmeans(lmer_simp)
plot(lmer_simp)
```

## Linear Mixed Effects Regression Analysis: Correlations

```{r lmer_predict_biomass, results = 'hide', eval = FALSE}
lmer_biomass_null <- lmer(biomass ~ site*nitrogen + (1|block:site), data = master, REML = FALSE)
lmer_biomass_rich <- lmer(biomass ~ site*nitrogen + div_rich + (1|block:site), data = master, REML = FALSE); anova(lmer_biomass_null, lmer_biomass_rich)
lmer_biomass_rar <- lmer(biomass ~ site*nitrogen + div_rar + (1|block:site), data = master, REML = FALSE); anova(lmer_biomass_null, lmer_biomass_rar)
lmer_biomass_shan <- lmer(biomass ~ site*nitrogen + div_shan + (1|block:site), data = master, REML = FALSE); anova(lmer_biomass_null, lmer_biomass_shan)
lmer_biomass_simp <- lmer(biomass ~ site*nitrogen + div_simp + (1|block:site), data = master, REML = FALSE); anova(lmer_biomass_null, lmer_biomass_simp)
```

```{r lmer_predict_yield, results = 'hide', eval = FALSE}
lmer_yield_null <- lmer(yield ~ site*nitrogen + (1|block:site), data = master, REML = FALSE)
lmer_yield_rich <- lmer(yield ~ site*nitrogen + div_rich + (1|block:site), data = master, REML = FALSE); anova(lmer_yield_null, lmer_yield_rich)
anova(lmer_yield_rich)
lmer_yield_rar <- lmer(yield ~ site*nitrogen + div_rar + (1|block:site), data = master, REML = FALSE); anova(lmer_yield_null, lmer_yield_rar)
lmer_yield_shan <- lmer(yield ~ site*nitrogen + div_shan + (1|block:site), data = master, REML = FALSE); anova(lmer_yield_null, lmer_yield_shan)
lmer_yield_simp <- lmer(yield ~ site*nitrogen + div_simp + (1|block:site), data = master, REML = FALSE); anova(lmer_yield_null, lmer_yield_simp)
```

```{r lmer_predict_height, results = 'hide', eval = FALSE}
lmer_height_null <- lmer(height ~ site*nitrogen + (1|block:site), data = master, REML = FALSE)
lmer_height_rich <- lmer(height ~ site*nitrogen + div_rich + (1|block:site), data = master, REML = FALSE); anova(lmer_height_null, lmer_height_rich)
lmer_height_rar <- lmer(height ~ site*nitrogen + div_rar + (1|block:site), data = master, REML = FALSE); anova(lmer_height_null, lmer_height_rar)
lmer_height_shan <- lmer(height ~ site*nitrogen + div_shan + (1|block:site), data = master, REML = FALSE); anova(lmer_height_null, lmer_height_shan)
lmer_height_simp <- lmer(height ~ site*nitrogen + div_simp + (1|block:site), data = master, REML = FALSE); anova(lmer_height_null, lmer_height_simp)
```

```{r lmer_predict_lodged, results = 'hide', eval = FALSE}
lmer_lodged_null <- lmer(lodged ~ site*nitrogen + (1|block:site), data = master, REML = FALSE)
lmer_lodged_rich <- lmer(lodged ~ site*nitrogen + div_rich + (1|block:site), data = master, REML = FALSE); anova(lmer_lodged_null, lmer_lodged_rich)
lmer_lodged_rar <- lmer(lodged ~ site*nitrogen + div_rar + (1|block:site), data = master, REML = FALSE); anova(lmer_lodged_null, lmer_lodged_rar)
lmer_lodged_shan <- lmer(lodged ~ site*nitrogen + div_shan + (1|block:site), data = master, REML = FALSE); anova(lmer_lodged_null, lmer_lodged_shan)
lmer_lodged_simp <- lmer(lodged ~ site*nitrogen + div_simp + (1|block:site), data = master, REML = FALSE); anova(lmer_lodged_null, lmer_lodged_simp)
anova(lmer_lodged_simp)
```

## Negative Binomial Model Analysis 

This analysis follows the [`phyloseq` tutorial](https://bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html) by Paul J. McMurdie and Susan Holmes.

```{r phyloseq}
info$nitrogen <- relevel(info$nitrogen, ref = "control")
physeq <- phyloseq(otu_table(asv, taxa_are_rows = FALSE), sample_data(info), refseq(fas), tax_table(tax))
```

```{r deseq}
physeq <- prune_samples(sample_sums(physeq) > 500, physeq)
dseq <- phyloseq_to_deseq2(physeq, ~ nitrogen + site)
gm_mean <- function(x, na.rm = TRUE){
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans <- apply(counts(dseq), 1, gm_mean)
dseq_geo = estimateSizeFactors(dseq, geoMeans = geoMeans)
dseq_geo = DESeq(dseq_geo, fitType="local")
```

```{r deseq_results}
res <- results(dseq_geo)
res <- res[order(res$padj, na.last = NA), ]
alpha <- 0.01
sigtab <- res[(res$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(physeq)[rownames(sigtab), ], "matrix"))
```


```{r pos_neg_significance_tables}
posigtab <- sigtab[sigtab[, "log2FoldChange"] > 0, ]
posigtab <- posigtab[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Order", "Family", "Genus")]
posigtab
negsigtab <- sigtab[sigtab[, "log2FoldChange"] < 0, ]
negsigtab <- negsigtab[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Order", "Family", "Genus")]
negsigtab
```

```{r all_sites_negative_binomial}
sigtabgen <- subset(sigtab, !is.na(Genus))
# Order
x <- tapply(sigtabgen$log2FoldChange, sigtabgen$Order, function(x) max(x))
x <- sort(x, TRUE)
sigtabgen$Order <- factor(as.character(sigtabgen$Order), levels = names(x))
# Genus
x <- tapply(sigtabgen$log2FoldChange, sigtabgen$Genus, function(x) max(x))
x <- sort(x, TRUE)
sigtabgen$Genus = factor(as.character(sigtabgen$Genus), levels = names(x))
ggplot(sigtabgen, aes(y = Genus, x = log2FoldChange, color = Order)) + 
  geom_vline(xintercept = 0.0, color = "gray", size = 0.5) +
  geom_point(size = 4) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust = 0.5))
```

## Ordination

```{r nmds, eval = FALSE}
# clean
physeq_trans <- transform_sample_counts(physeq, function(x) 1E6 * x/sum(x))
# plot ASVs
physeq_trans_ord <- ordinate(physeq_trans, "NMDS", "bray")
p1 <- plot_ordination(physeq_trans, physeq_trans_ord, type="taxa", color="Order", title="taxa")
print(p1)
# plot samples
p2 <- plot_ordination(physeq_filt_trans, physeq_filt_trans_ord, type="samples", color="site", title="Nirogen")
print(p2)
```

```{r nmds_filt}
# clean
physeq
filt <- genefilter_sample(physeq, filterfun_sample(function (x) x > 5))
physeq_filt <- prune_taxa(filt, physeq)
physeq_filt
physeq_filt_trans <- transform_sample_counts(physeq_filt, function(x) 1E6 * x/sum(x))
physeq_filt_trans
# plot ASVs
physeq_filt_trans_ord <- ordinate(physeq_filt_trans, "NMDS", "bray")
p1 <- plot_ordination(physeq_filt_trans, physeq_filt_trans_ord, type="taxa", color="Genus", title="taxa")
print(p1)
# plot samples
p2 <- plot_ordination(physeq_filt_trans, physeq_filt_trans_ord, type="samples", color="site", title="NMDS Ordination of ASV Communities", shape = "nitrogen")
print(p2)
p3 <- p2 + geom_line(aes(group = site:block))
print(p3)
```

```{r nmds_filt_biplot, eval = FALSE}
p3 <- plot_ordination(physeq_filt_trans, physeq_filt_trans_ord, type="biplot", color="site", shape="Family", title="biplot")
# Some stuff to modify the automatic shape scale
GP1.shape.names = get_taxa_unique(physeq_filt_trans, "Family")
GP1.shape <- 15:(15 + length(GP1.shape.names) - 1)
names(GP1.shape) <- GP1.shape.names
GP1.shape["samples"] <- 16
p3 + scale_shape_manual(values=GP1.shape)

p4 <- plot_ordination(physeq_filt_trans, physeq_filt_trans_ord, type="split", color="Family", shape="nitrogen", label="site", title="split") 
p4
```

